### DISCUSSION TITLE
Alpha ordering of json keys

### DISCUSSION THREAD

**Discussion by adrianboston** [taocpp/json#77] [Q&A] [answered]
toajson outputs a json string alphabetically ordered by key -- opposed to the order a document was constructed. 
is this a function of the to_string function.
tao::json::to_string( document );

was this a particular design decision. 
and the question : can it be output as constructed.

**Comment by ColinH** [comment, marked as answer]
When your document is stored in-memory as `tao::json::value` then there is no direct way to restore the insertion order of object members given that they are held in a `std::map< std::string, tao::json::value >` which sorts them by key.

If you can avoid the intermediate step of creating a `tao::json::value`, and directly generate JSON Events instead, you could use `tao::json::events::to_string` as they directly generate the output from the Events and don't care about order.

Otherwise I don't see any easy solution, you could add a custom base class to the values in the object, store the insertion order there, and somehow use it later, but that would probably be quite cumbersome depending on how the value is created.

**Comment by adrianboston** [comment]
ah. the map. of course.
it might be done with a boost multiindex but..its not important. its really for human exam. i was expecting a different output like that found w rapidjson. otherwise very legible and easy to use library. thank you
